### PR TITLE
Compat.eachline with keep kwarg

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

**Comment by fredrikekre** [comment]
Nightly failure:
```
Test Failed at /home/travis/.julia/v0.7/Compat/test/runtests.jl:783
  Expression: [a for (a, b) = pairs(IndexLinear(), A14)] == [1, 2, 3, 4]
   Evaluated: [1 3; 2 4] == [1, 2, 3, 4]
```
probably related to https://github.com/JuliaLang/julia/pull/26775 @nalimilan.

**Comment by martinholters** [comment]
Test failure is #540.

File: README.md [hunk 1]
```diff
@@ -180,6 +180,8 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 * `Compat.readline` with `keep` keyword argument ([#25646])
 
+* `Compat.eachline` with `keep` keyword argument ([#25646])
```
Comment (nalimilan) [inline README.md]: Merge with previous line? Otherwise, LGTM.

Comment (fredrikekre) [reply README.md]: Leaving the previous line untouched makes it simpler to find Compat version with git blame

Comment (nalimilan) [reply README.md]: As you want.

Comment (ararslan) [reply README.md]: I don't understand. They refer to the same PR and have identical text. Why was there a line already there for it?

Comment (fredrikekre) [reply README.md]: > identical text

`readline != eachline`

Comment (ararslan) [reply README.md]: WOW. :man_facepalming: Whoops. I agree then, separate lines probably makes more sense here.

**Pull request merged**